src/datalink/mod.rs

```diff
@@ -229,6 +229,15 @@ impl NetworkInterface {
     }
 }
 
+/// Represents an IP address and subnet mask
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub struct IpNetmask {
+    /// IP address
+    pub ip: IpAddr,
+    /// Subnet mask
+    pub netmask: Option<IpAddr>,
```

Comment:
"if applicable for the address family" seems to be the key phrase to me - if the interface has an IPv4 or IPv6 address, a netmask will always be applicable (even if it's `0.0.0.0`/`::`).